DomAssignment/validations.js

```diff
@@ -0,0 +1,234 @@
+var mobile_reg = /^(\+91|\+91-|0)?\d{10}$/; // Validating Phone Number according to India
+
+
+
+function register() {
+
+
+
+	if (firstname === "") {
+		alert("Enter FirstName");
+		return false;
+	}
+	if (!firstname_reg.test(firstname)) {
+		alert("First Name is Invalid");
+		return false;
+	}
+	if (paddress === "") {
+
+		alert("Enter Parmanent Address");
+		return false;
+	}
+
+	if (caddress === "") {
+
+		alert("Enter Current Address");
+		return false;
+	}
+	if (email === "") {
+		alert("Enter Email");
+		return false;
+
+	}
+	if (!mailformat.test(email)) {
+		alert("Email Invalid");
+		return false;
+	}
+
+	if (phone_no === "") {
+
+		alert("Enter Phone Number");
+		return false;
+	}
+	if (!mobile_reg.test(phone_no)) {
+		alert("Enter Valid Phone Number");
+		return false;
+
+	}
+	if (password === "") {
+		alert("Enter Password");
+		return false;
+	}
+	if (cpassword === "") {
+		alert("Please Confirm Password");
+		return false;
+	}
+	if (!reg_password.test(password)) {
+		alert("Please Add Valid Password");
+		return false;
+	}
+	if (password != cpassword) {
+		alert("Password Not Same");
+		return false;
+
+	} else {
+
+
+		var parentElement = document.getElementById("parent-right");
+		var childElement = document.createElement("input");
+		childElement.setAttribute("id", "child-button");
+		childElement.setAttribute("title", email);
+		childElement.setAttribute("value", firstname + " " + lastname);
+		childElement.setAttribute("type", "button");
+		childElement.setAttribute("class", "inputedit");
+		parentElement.appendChild(childElement);
+
+
+		var frm = document.getElementsByClassName("Registration_Form");
+		frm.reset();
+		alert("Registered");
+
+	}
+
+
+
+}
+
+
+function fillAddress(f) {
+
+	if (f.checkboxtype.checked == true) {
+
+		f.caddress.value = f.paddress.value;
+		document.getElementById("cadd").disabled = true;
+
+	}
+	if (f.checkboxtype.checked == false) {
+
+		document.getElementById("cadd").disabled = false;
+
+	}
+
+
+
+}
+
+
+function firstname() {
+
+	if (firstname === "") {
+		alert("Enter FirstName");
+		return false;
+	}
+
+}
+
+function p_address() {
+
+	if (paddress === "") {
+
+		alert("Enter Parmanent Address");
+		return false;
+	}
+
+}
+
+function c_address() {
+
+	if (caddress === "") {
+
+		alert("Enter Current Address");
+		return false;
+	}
+
+}
+
+function email() {
+
+
+	if (email === "") {
+		alert("Enter Email");
+		return false;
+
+	}
+
+
+}
+
+
+function phone() {
+
+	if (phone_no === "") {
+
+		alert("Enter Phone Number");
+
+
+	}
+
+}
+
+function e_password() {
+
+
+	if (password === "") {
+		alert("Enter Password");
+		return false;
+	}
+
+}
+
+function e_password() {
+	if (cpassword === "") {
+		alert("Please Confirm Password");
+		return false;
+	}
+
+}
+
+
+
+window.onclick = function(event) {
+	if (!event.target.matches('.dropbtn')) {
+
+		var dropdowns = document.getElementsByClassName("dropdown-content");
+		var i;
+		for (i = 0; i < dropdowns.length; i++) {
```

Comment:
do u really need a for loop here?
this can be done by removing show class of the dropdown-container